doc/ensemble.rst

```diff
@@ -0,0 +1,739 @@
+
+-  `name`: "cnn" - A key to uniquely identify the model within the database.
+-  `backend`: "TF" - Indicating that the model is a TensorFlow model.
+-  `model`: model - The in-memory representation of the TensorFlow model.
+-  `device`: "CPU" - Specifying the device for ML model execution.
+-  `inputs`: inputs - The names of the model inputs.
+-  `outputs`: outputs - The names of the model outputs.
+
+When the ensemble is started via ``Experiment.start()``, the ML model will be loaded to the
+standard orchestrator that is launched prior to the start of the ensemble.
+
+---------------------
+TorchScript functions
+---------------------
+.. _load_tf_function:
+Users can instruct SmartSim to upload TorchScript functions to the database
+at runtime. Script functions are loaded into
+standard orchestrators prior to the execution of ensemble entities. If using a
+colocated orchestrator, use the ``Ensemble.add_script()`` function.
+Users have the flexibility to choose
+between `"GPU"` or `"CPU"` for device selection, and in environments with multiple
+devices, specific device numbers can be specified via `devices_per_node`.
+
+When specifying a TF function using ``Ensemble.add_function()``, the
+following arguments are offered:
+
+-  `name`  (str) : key to store function under
+-  `function` (str, optional) : Name of the TorchScript function
+-  `device`  (str, optional) : device for script execution, defaults to “CPU”
+-  `devices_per_node` (int) : assign the number of CPU's or GPU's to use on the node
+
+Example: Loading an TensorFlow Function to the Ensemble
+-------------------------------------------------------
+In this example, we demonstrate how to instruct SmartSim to load
+an TensorFlow function into the database at ensemble runtime. It's
+important to note the function, ``Ensemble.add_function()`` is supported
+for non-colocated deployments and during a colocated deployment, ``Ensemble.add_script()``
+should be used.
+
+**Python Script: Define a TF Function for Ensemble Purposes**
+To load a TF function, define the function within the Python driver script.
+
+.. code-block:: python
+
+    def timestwo(x):
+        return 2*x
+
+**SmartSim Ensemble Integration:**
+
+Assuming an initialized ``Ensemble`` named `smartsim_ensemble`, we specify the
+following parameters to the ``Ensemble.add_function()`` function:
+
+.. code-block:: python
+
+    smartsim_ensemble.add_function(name="example_func", function=timestwo, device="CPU")
+
+In this integration, we provide the following details:
+
+-  `name`: "example_func" - A key to uniquely identify the model within the database.
+-  `function`: timestwo - Name of the TorchScript function defined in the Python driver script.
+-  `device`: "CPU" - Specifying the device for ML model execution.
+
+When the ensemble is started via ``Experiment.start()``, the TF function will be loaded to the
+standard orchestrator that is launched prior to the start of the ensemble.
```

Comment:
we said in opening that it's going to run on both colo & standard but imply that it only runs on standard here

Reply:
i see that this example is consuming `add_function` not `add_script` and the result will run on standard orch. please disregard!